src/StaticATokenLM.sol

```diff
@@ -485,7 +486,10 @@ contract StaticATokenLM is
     uint16 referralCode,
     bool fromUnderlying
   ) internal returns (uint256) {
+    require(assets != 0, StaticATokenErrors.INVALID_ZERO_AMOUNT);
     require(recipient != address(0), StaticATokenErrors.INVALID_RECIPIENT);
+    uint256 shares = previewDeposit(assets);
+    require(shares != 0, StaticATokenErrors.INVALID_ZERO_AMOUNT);
```

Comment:
This should take care of the case where a user asks to deposit anywhere between [0-0.5) AToken worth of underlying.
The require will make sure to revert in such a case.